packages/react/src/xcss-prop/__tests__/media-query-strict.test.tsx

```diff
@@ -0,0 +1,72 @@
+/** @jsxImportSource @compiled/react */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { render } from '@testing-library/react';
+
+import { cssMap } from '../../create-strict-api/__tests__/__fixtures__/strict-api';
+import type { XCSSProp } from '../../create-strict-api/__tests__/__fixtures__/strict-api';
+import type { XCSSAllProperties, XCSSAllPseudos } from '../index';
+
+function CSSPropComponent({
+  xcss,
+}: {
+  xcss: ReturnType<typeof XCSSProp<XCSSAllProperties, XCSSAllPseudos>>;
+}) {
+  return <div className={xcss}>foo</div>;
+}
+
+const styles = cssMap({
+  invalid: {
+    // @ts-expect-error — @media at rule object is not allowed in strict cssMap
+    '@media': {
+      screen: { color: 'red' },
+    },
+  },
+  valid: {
+    '@media (min-width: 110rem)': {
+      // TODO: This should be a type error as color should be one of the base styles.
+      color: 'red',
+    },
+  },
+});
+
+describe('xcss prop', () => {
+  it('should allow valid media queries in inline xcss prop', () => {
+    const { getByText } = render(
+      <CSSPropComponent
+        xcss={{
+          '@media (min-width: 110rem)': {
+            // TODO: This should be a type error as color should be one of the base styles.
```

Comment:
TODO 👀 in the future?

Reply:
TODO right now. I'm currently working on another PR to merge before this one that fixes up the types.